AppImage/make-appimage.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh
+
+# This script is meant to be ran on alpine linux, dependencies:
+# wget build-base patchelf libxscrnsaver-dev libxinerama-dev libxrandr-dev libnotify-dev
+# dbus-dev wayland-dev perl pango-dev wayland-protocols dbus librsvg desktop-file-utils
+
+set -e
+export ARCH="$(uname -m)"
+export APPIMAGE_EXTRACT_AND_RUN=1
+APPIMAGETOOL="https://github.com/AppImage/appimagetool/releases/download/continuous/appimagetool-x86_64.AppImage"
+export COMPLETIONS=0
+
+# BUILD DUNST
+mkdir -p ./AppDir
+make -j$(nproc) PREFIX="$PWD"/AppDir/usr
+make install PREFIX="$PWD"/AppDir/usr
+cd ./AppDir
+
+# AppRun
+echo "Creating AppRun..."
+cat >> ./AppRun << 'EOF'
+#!/bin/sh
+CURRENTDIR="$(dirname "$(readlink -f "$0")")"
+export PATH="$CURRENTDIR/usr/bin:$PATH"
+[ -z "$APPIMAGE" ] && APPIMAGE="$0"
+BIN="${ARGV0#./}"
+unset ARGV0
+if [ -f "$CURRENTDIR/usr/bin/$BIN" ]; then
+	if [ "$BIN" = "dunstctl" ]; then
+		exec "$CURRENTDIR/usr/bin/$BIN" "$@"
+	else
+		exec "$CURRENTDIR/ld-musl-x86_64.so.1" "$CURRENTDIR/usr/bin/$BIN" "$@"
+	fi
+elif [ "$1" = "--notify" ]; then
+	shift
+	exec "$CURRENTDIR/ld-musl-x86_64.so.1" "$CURRENTDIR"/usr/bin/dunstify "$@"
+elif [ "$1" = "--ctl" ]; then
+	shift
+	exec "$CURRENTDIR"/usr/bin/dunstctl "$@"
+else
+	if [ -z "$1" ]; then
+		echo "AppImage commands:"
+		echo " \"$APPIMAGE\"            runs dunst"
+		echo " \"$APPIMAGE --notify\"   runs dunstify"
+		echo " \"$APPIMAGE --ctl\"      runs dunstctl"
+		echo "You can also make and run symlinks to the AppImage with the names"
+		echo "dunstify and dunstctl to launch them automatically without extra args"
+		echo "running dunst..."
+	fi
+	exec "$CURRENTDIR/ld-musl-x86_64.so.1" "$CURRENTDIR/usr/bin/dunst" "$@"
+fi
```

Comment:
Why don't you run with `$APPIMAGE <dunst|dunstify|dunstctl> [options]` and ignore the handling of `--notify` and `--ctl` ?

This would be a simple switch case then:

```suggestion
case in "${BIN}" in
    dunst|dunstify)
        exec "$CURRENTDIR/ld-musl-${ARCH}.so.1" "$CURRENTDIR/usr/bin/$BIN" "$@"
        ;;
    dunstctl)
        exec "$CURRENTDIR/usr/bin/$BIN" "$@"
        ;;
    "")
        echo "AppImage commands:"
        echo " \"$APPIMAGE dunst\"      runs dunst"
        echo " \"$APPIMAGE dunstify\"   runs dunstify"
        echo " \"$APPIMAGE dunstctl\"   runs dunstctl"
        echo "You can also make and run symlinks to the AppImage with the names"
        echo "dunstify and dunstctl to launch them automatically without extra args"
        echo "running dunst..."
        "${APPIMAGE}" dunst
        ;;
    *)
        echo "AppImage commands:"
        echo " \"$APPIMAGE dunst\"      runs dunst"
        echo " \"$APPIMAGE dunstify\"   runs dunstify"
        echo " \"$APPIMAGE dunstctl\"   runs dunstctl"
        echo "You can also make and run symlinks to the AppImage with the names"
        echo "dunstify and dunstctl to launch them automatically without extra args"
        ;;
esac
```

_I acknowledge, that I have copied the other bashism `"$@"`, since I expect you'll switch shells to bash._

Also the `echo` commands could be optimized further.